backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/IShipmentScheduleEffectiveBL.java

```diff
@@ -66,6 +66,9 @@ public interface IShipmentScheduleEffectiveBL extends ISingletonService
 	/** Consider using {@link IShipmentScheduleBL#getQtyToDeliver(I_M_ShipmentSchedule)} instead. */
 	BigDecimal getQtyToDeliverBD(I_M_ShipmentSchedule sched);
 
+	@Nullable
+	BigDecimal getQtyOnHand(@NonNull final I_M_ShipmentSchedule sched);
```

Comment:
please, use Quantity instead of BigDecimal since you have access to the C_UOM_ID as well.